File: 7-kyu-kata/sort-arrays-1/python/sort_arrays_1_test.py
# 7 kyu - Sort arrays - 1  [ ID: 51f41b98e8f176e70d0002a8  (sort-arrays-1) ]
# URL: https://www.codewars.com/kata/51f41b98e8f176e70d0002a8
# Category: REFERENCE  |  Tags: SORTING | ARRAYS | FUNDAMENTALS
# ******************************************************************************
import codewars_test as test
from solution import sortme

@test.describe("Fixed Tests")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(sortme(["one", "two", "three"]), ["one", "three", "two"])

@test.describe("Random Tests")
def random_tests():
    
    from random import randint, sample
      
    for _ in range(100):          
        arr = ["".join(sample("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz", k = randint(4, 10))) for i in range(randint(3,6))]
        expected = sorted(arr)
        @test.it(f"testing for sortme({arr})")
        def test_case():
            test.assert_equals(sortme(arr[:]), expected)

File: 7-kyu-kata/sort-arrays-1/javascript/sortArrays1.Test.js
// 7 kyu - Sort arrays - 1  [ ID: 51f41b98e8f176e70d0002a8  (sort-arrays-1) ]
// URL: https://www.codewars.com/kata/51f41b98e8f176e70d0002a8
// Category: REFERENCE  |  Tags: SORTING | ARRAYS | FUNDAMENTALS
// *****************************************************************************
const { assert } = require("chai")

describe("Basic tests", () => {
  it("Testing for fixed tests", () => {
    assert.deepEqual(sortme(["one", "two", "three"]), ["one", "three", "two"])
  })
})

describe("Random tests", () => {
  const s = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"

  function randint(min, max) {
    return Math.floor(Math.random() * (max - min + 1)) + min
  }

  for (let i = 0; i < 100; i++) {
    let arr = Array.from(
      { length: randint(3, 6) },
      (_, i) => Array.from({ length: randint(4, 10) }, (_, i) => s[randint(0, s.length - 1)]).join``
    )
    let expected = arr.sort()
    it(`sortme(${JSON.stringify(arr)}) should equal ${JSON.stringify(expected)}`, () => {
      assert.deepEqual(sortme(arr.slice()), expected)
    })
  }
})
